### ISSUE TITLE
Eslint and vuese default properties

### ISSUE THREAD

**Issue by VitoriaFaria** [HcySunYang/vuese#76]
**Describe the bug**
The default declaration to an array property in Eslint is
`default () { return [] }`
like they describe in this link https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/require-valid-default-prop.md,
but when i use vuese my output is the below because vuese search for 
`default: []`

**Vuese output pictures**

![Screenshot_2019-05-21 TraysCard - Components](https://user-images.githubusercontent.com/40899056/58112585-325bd480-7bca-11e9-92ff-961ecfb29183.png)

**Comment by HcySunYang** [comment]
Which version are you using?

**Comment by VitoriaFaria** [comment]
vuese/2.3.1

**Comment by HcySunYang** [comment]
@VitoriaFaria Different types of props have different ways of handling their default values. The current temporary solution is to manually add annotations to override the default behavior of `vuese`:

```js
props: {
    a: {
      type: Array,
      // `[]`
      default () {
        return []
      }
    }
}
```

**Comment by VitoriaFaria** [comment]
Yeah, I did it and works, but is sad that vuese don't recognise this kind of code. However, it's a amazing project, congratulations for your hardworking. :+1:  :100:  :1st_place_medal:

**Comment by flyinglove** [comment]
but what if the default value is a object with many keys, the markdown file will generate normally, but then render ui is a little mess, like this: 
![image](https://user-images.githubusercontent.com/12018559/62522612-9b1b2a80-b864-11e9-9585-25e5198ef9fe.png)

**Comment by HcySunYang** [comment]
My idea is that if the type of a prop is a reference type, then the default function is executed to get the default value, but this actually causes a lot of problems.

**Comment by flyinglove** [comment]
> My idea is that if the type of a prop is a reference type, then the default function is executed to get the default value, but this actually causes a lot of problems.

I think it's reasonable to get the default value like this, and this project is great,  i am wondering if it's possible to format the default value if it's complex

**Comment by HcySunYang** [comment]
Internally, we use babel to get the "string code" of the `default` function, then create a function with `new Function` and execute it. Because this process is not fully runtime, so we may encounter unknown errors, such as using externally defined variables or constants in the `default` function.

**Comment by flyinglove** [comment]
i haven't seen any code using external variable or defined constants in `default` function,  and does this have anything to do with my question? 😭

**Comment by HcySunYang** [comment]
I think this is the difficulty of writing markdown. 😄

**Comment by daniegarcia254** [comment]
I would say it is related to this issue, so for now, instead of open a new issue, I'll just put it here.

When you use an Object property, an set de default value:
```
    position: {
      type: Object,
      default: () => ({
        vertical: "RIGHT",
        horizontal: "TOP"
      }),
      required: false
    }
```
and generate doc with vuese, it reports an error:

> The error occurred when processing: /Users/danielgarcia/devel/repos/modules/packages/case-vue-ui-components/src/components/VueOptions.vue
🚨  SyntaxError: Unexpected token :
    at new Function (<anonymous>)
    at runFunction (/.../modules/node_modules/vuese/dist/vuese.js:54:16)
    at otherNodes.forEach (/.../node_modules/vuese/dist/vuese.js:382:38)
    at Array.forEach (<anonymous>)
    at processPropValue (/.../node_modules/vuese/dist/vuese.js:378:20)
    at ObjectProperty (/.../node_modules/vuese/dist/vuese.js:206:41)
    at NodePath._call (/.../node_modules/@babel/traverse/lib/path/context.js:53:20)
    at NodePath.call (/.../node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/...node_modules/@babel/traverse/lib/path/context.js:88:12)

and the doc for this component `VueOptions.vue` it is not generated, when the use of defaults is correct, so it shouldn't fail.

If we change the defaults to:
```
    position: {
      type: Object,
      default: () => {
        return {
          vertical: "RIGHT",
          horizontal: "TOP"
        }
      },
      required: false
    }
```
It works okey, but in the generated doc, the default value for this prop will show `[object Object]` instead of the default values `{ vertical: "RIGHT", horizontal: "TOP" }`

I've made some research, and as @HcySunYang has pointed, it is related to `@babel/generator` when it generates de function code. ¿Do you think is possible to fix this in anyway? It is a problem if I cannot generate the documentation for a whole component.

**Comment by HcySunYang** [comment]
My idea is that if vuese encounters an error while executing the `default` function, the default value is ignored instead of throwing an error.  In addition, you can document the default values ​​by adding annotations.

**Comment by HcySunYang** [comment]
I just released `@vuese/cli@2.7.1` to solve the following problem.
> It works okey, but in the generated doc, the default value for this prop will show [object Object] instead of the default values { vertical: "RIGHT", horizontal: "TOP" }

@daniegarcia254
